org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java

```diff
@@ -282,6 +313,62 @@ public LanguageServerProjectExecutor excludeInactive() {
 		}
 	}
 
+	/**
+	 * Specialised POJO for document edits specifically
+	 *
+	 */
+	public static class VersionedEdits extends Versioned<List<? extends TextEdit>> {
```

Comment:
Can we extract the class `VersionedEdits` out of `LanguageServers` into `LSPFormatter`? I think we should keep this class generic.

Reply:
I moved it here because `VersionedEdits` *is* generic: other operations can produce edits so would benefit from this (eventually, when they are ported over...)

Having it in the formatter class ties it to formatting, and it isn't really specific to formatting.

I did ponder making it a top-level class rather than a nested class within `LanguageServers`. What do you think?

Reply:
Yes, I think it makes sense.